tdms/src/fields/td_field_exporter_2d.cpp

```diff
@@ -0,0 +1,39 @@
+#include "matlabio.h"
+#include "field.h"
+
+
+void TDFieldExporter2D::allocate(int I, int J) {
+
+  mwSize dims[2] = {I, J};
+  matlab_array = mxCreateNumericArray(2, (const mwSize *)dims, mxDOUBLE_CLASS, mxREAL);
+  array = castMatlab2DArray(mxGetPr((mxArray *) matlab_array), I, J);
+}
+
+TDFieldExporter2D::~TDFieldExporter2D() {
+  freeCastMatlab2DArray(array);
+}
+
+
+void TDFieldExporter2D::export_field(SplitField& F, int stride, int iteration) const{
+
+  int ic = 0;
+
+  for (int i = 0; i < F.I_tot; i++) {
+    int kc = 0;
+    if ((i % stride) == 0) {
```

Comment:
Instead of comparing every value of `i` here, it would be clearer to do something like:
```
int i = 0;
while i < F.I_tot {
    // do logic
   i += stride;
}
```
and similarly for `k`.